test/jdk/javax/swing/JInternalFrame/InternalFrameBorderTest.java

```diff
@@ -0,0 +1,248 @@
+            default -> throw new IllegalStateException("Unexpected value: "
+                    + borderDirection);
+        }
+
+        boolean isVertical =  borderDirection.equals("RIGHT")
+                || borderDirection.equals("LEFT");
+        boolean isHorizontal =  borderDirection.equals("TOP")
+                || borderDirection.equals("BOTTOM");
+
+        robot.mouseMove(x, y);
+        for (int i = start; i < stop; i++) {
+            if (Color.RED.equals(robot.getPixelColor(
+                    isVertical ? i : (iFrameLoc.x + MIDPOINT),
+                    isHorizontal ? i : (iFrameLoc.y + MIDPOINT)))) {
+                        saveScreenCapture(borderDirection + "_" + uiScale + ".png");
+                        errorLog.append("uiScale: "+ uiScale +
+                                " Red background color" + " detected at "
+                                + borderDirection + " border\n");
+            }
+        }
+        robot.delay(300);
+    }
+
+    private static void checkCorners(String cornerLocation) {
+        int x, y = 0;
+
+        switch (cornerLocation) {
+            case "TOP_LEFT" -> {
+                x = iFrameLoc.x;
+                y = iFrameLoc.y;
+            }
+            case "TOP_RIGHT" -> {
+                x = iFrameMaxX;
+                y = iFrameLoc.y;
+            }
+            case "BOTTOM_RIGHT" -> {
+                x = iFrameMaxX;
+                y = iFrameMaxY;
+            }
+            case "BOTTOM_LEFT" -> {
+                x = iFrameLoc.x;
+                y = iFrameMaxY;
+            }
+            default -> throw new IllegalStateException("Unexpected value: "
+                    + cornerLocation);
+        }
+        robot.mouseMove(x, y);
+        if (Color.RED.equals(robot.getPixelColor(x, y))) {
+            saveScreenCapture(cornerLocation + "_" + uiScale + ".png");
+            errorLog.append("uiScale: "+ uiScale + " Red background color" +
+                    " detected at " + cornerLocation +  " corner\n");
+        }
+        robot.delay(300);
+    }
+
+    private static void createAndShowGUI() {
+        jFrame = new JFrame();
+        jFrame.setSize(FRAME_SIZE, FRAME_SIZE);
+        jFrame.setLayout(null);
+        jFrame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+        JLabel scale = new JLabel("UI Scale: "+ uiScale);
+
+        iFrame = new JInternalFrame("iframe", true);
+        iFrame.setLayout(new GridBagLayout());
+        iFrame.setBackground(Color.RED);
+        iFrame.add(scale);
+        iFrame.setLocation(30, 30);
+        jFrame.getContentPane().add(iFrame);
+        iFrame.setSize(INTFRAME_SIZE, INTFRAME_SIZE);
+        iFrame.setVisible(true);
+
+        jFrame.setLocation(150, 150);
+        jFrame.setVisible(true);
+    }
+    // for debugging purpose, saves screen capture when test fails.
+    private static void saveScreenCapture(String filename) {
+        BufferedImage image = robot.createScreenCapture(jFrame.getBounds());
```

Comment:
I guess we lose pixel details here. If uiScale > 1, `createScreenCapture` scales down the captured image. Do we want to preserve the original image too? [`Robot.createMultiResolutionScreenCapture`](https://docs.oracle.com/en/java/javase/17/docs/api/java.desktop/java/awt/Robot.html#createMultiResolutionScreenCapture(java.awt.Rectangle)) can be useful.